include/mqt-core/operations/Operation.hpp

```diff
@@ -171,6 +182,18 @@ class Operation {
                             const RegisterNames& creg, size_t indent,
                             bool openQASM3) const = 0;
 
+  /// Checks whether operation commutes with other operation on a given qubit.
+  [[nodiscard]] virtual auto
+  commutesAtQubit(const Operation& /*other*/,
+                  const Qubit& /*qubit*/) const -> bool {
+    return false;
+  }
+
+  [[nodiscard]] virtual auto
+  isInverseOf(const Operation& /*other*/) const -> bool {
+    return false;
+  }
```

Comment:
No, I can't. This function is actually work-in-progress. The reason why your suggestion does not work is, that two compound operations with a 1-Q operation on each qubit might be inverse of each other but the order of the operations is not reversed (that happens for my circuits when I read them in). Then this approach would return false when I need it to return true.

Reply:
Ah yeah. One would have to topologically sort the inverse in order to get a canonical ordering and make that conclusion. It might still be possible to make that work somehow without duplicating too much code.

Reply:
In the next commit that I will push later, I try this suggestion together with a special handling in the compound operation. I mean for non-CompouindOperations this should already work fine just that it might not the most performant approach to first generate the inverse and then to check that the operand is equivalent especially in cases where it is not the inverse.